hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/BlockingRpcConnection.java

```diff
@@ -257,7 +256,17 @@ protected void setupConnection() throws IOException {
         if (this.rpcClient.localAddr != null) {
           this.socket.bind(this.rpcClient.localAddr);
         }
-        NetUtils.connect(this.socket, remoteId.getAddress(), this.rpcClient.connectTO);
+        if (this.rpcClient.metrics != null) {
```

Comment:
This change is a forward port of the branch-1 change here: https://github.com/apache/hbase/pull/2671/files#diff-1a7ec27a8107293b6c87132823c262fc250570687a40f45646c43ae46dc6b04eR255

This change fixes a bug we encountered in production while running in Amazon's Elastic Kubernetes Service.